dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java

```diff
@@ -129,7 +141,15 @@ private boolean checkDisabled(Map<String, String> oTelConfigs) {
     }
 
     private HashMap<String, String> getTelemetryProperties() {
+
         HashMap<String, String> telemetryProperties = new HashMap<>();
+
+        String defaultServiceName = lookupAppName();
+
+        if (defaultServiceName != null) {
+            telemetryProperties.put("otel.service.name", defaultServiceName);
+        }
```

Comment:
It's probably better to use [addResourceCustomizer](https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-extension-autoconfigure/1.22.0-alpha/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.html#addResourceCustomizer(java.util.function.BiFunction)) rather than setting default config.

It gives you control over whether you are overriding config or not and avoids config edge cases (e.g. if OTel checks `otel.service.name` before `OTEL_SERVICE_NAME` there's a chance that it picks up your default rather than `OTEL_SERVICE_NAME` defined in the server environment.